airflow/api_connexion/endpoints/xcom_endpoint.py

```diff
@@ -49,22 +52,28 @@ def get_xcom_entries(
     """
 
     query = session.query(XCom)
-    if dag_id != '~':
+    if dag_id == '~':
+        appbuilder = current_app.appbuilder
+        readable_dag_ids = appbuilder.sm.get_readable_dag_ids(g.user)
+        query = query.filter(XCom.dag_id.in_(readable_dag_ids))
+        query.join(DR, and_(XCom.dag_id.in_(readable_dag_ids), XCom.execution_date == DR.execution_date))
+    else:
         query = query.filter(XCom.dag_id == dag_id)
```

Comment:
When there's a single dag_id the permissions decorator will validate access to that dag, correct?

Reply:
Correct